2.5.2 Changes

IceDragon200 opened the pull request:
Projected fixes and changes for 2.5.2

Ignore the version number, it will be bumped later
## Builds
### Latest

**2016-04-06 @ 18:59:00 EST** 6b41a7e [2.5.1 - Download](https://dl.dropboxusercontent.com/u/72618186/minecraft/mods/growthcraft/2.5.1/growthcraft-1.7.10-2.5.1-complete--6b41a7e.jar)

> #### Checksums
> 
> **MD5**: 9531d2c6dfc1bae3cc5e6e3b907a882e
> **SHA1**: e4f5c86dbd844ae50a5e06910574ee8aac63399b
> **SHA256**: a4d51b534a00e28f9bcfd17573111dadcec5d83cafb7696ea2677e6f9b6aedfe
### Old

**2016-04-05 @ 17:41:14 EST** f6e4415 [2.5.1 - Download](https://dl.dropboxusercontent.com/u/72618186/minecraft/mods/growthcraft/2.5.1/growthcraft-1.7.10-2.5.1-complete--f6e4415.jar)

> #### Checksums
> 
> **MD5**: e7cd605c45dd6faceac486a5b56624bd
> **SHA1**: 81e5e90c5ee9fbfe6b1922d420dfef97898814d0
> **SHA256**: 460a963ac44164dbd024bf47f7e873d5c8733a6cd300dec31b755f3664eda4a2

**2016-04-05 @ 15:02:53 EST** fac4b19 [2.5.1 - Download](https://dl.dropboxusercontent.com/u/72618186/minecraft/mods/growthcraft/2.5.1/growthcraft-1.7.10-2.5.1-complete--fac4b19.jar)

> #### Checksums
> 
> **MD5**: 292391561e9a6b8567deee6b5fb2ca1c
> **SHA1**: ba83a86fc6d4312a7cb77b4702c8c3c52c6f7467
> **SHA256**: 24ddd3c4bc88d9c0cf358256ddaeae60c4e59a08c586e4ccf36209d311ccc4bd

**2016-04-05 @ 09:59:55 EST** 9b3e49f  [2.5.1 - Download](https://dl.dropboxusercontent.com/u/72618186/minecraft/mods/growthcraft/2.5.1/growthcraft-1.7.10-2.5.1-complete--9b3e49f.jar)

> #### Checksums
> 
> **MD5**: e5d64988e1aea945d49a1e5915cafec6
> **SHA1**: 4ce1c863119944bbe5292968dc9ca82597b71ec0
> **SHA256**: fccafd1437a79810888c9494888b79efcb7c91687f3675219b0e0f9a4c7d6af0

**2016-04-05 @ 01:12:37 EST** 586d6ab [2.5.1 - Download](https://dl.dropboxusercontent.com/u/72618186/minecraft/mods/growthcraft/2.5.1/growthcraft-1.7.10-2.5.1-complete--586d6ab.jar)

> #### Checksums
> 
> **MD5**: 95520d8ea7e93979b1b4103cdce20e06
> **SHA1**: 85415194fb19d171e93cf2bb7ad42932cc75ce6c
> **SHA256**: b4d709fe8b95a00f3ff9329ae29cb2ec42dd70bdb4353937547d55e4a4f3fa5d

IceDragon200 committed:
bamboo: Register bamboo fence rope as flammable

IceDragon200 committed:
core: Added fence ropes for Et Futurum fences

IceDragon200 committed:
api/cellar,api/core,core,milk: Code gardening

IceDragon200 committed:
Merge pull request #302 from GrowthcraftCE/etfuturum-fence-ropes

core: Added fence ropes for Et Futurum fences

IceDragon200 committed:
api/cellar,api/core,cellar,core,rice: Added IMultiItemStack support to brewing recipes

IceDragon200 committed:
api/cellar: Fruit Press should now ignore NBT data

IceDragon200 committed:
core: A bit more work on Fence Ropes

IceDragon200 committed:
api/cellar,apples,cellar: Fermenting and Pressing recipes support IMultiItemStacks

IceDragon200 committed:
core: Added Natura fence ropes (untested)

Sunconure11 commented:
You might want to talk to the Dynamic Surroundings dev, to see what he did to pick up on ore dictionaries, so recipes can boot up upon loading a world. The issue is partially fixed, but I still have issues with a custom definition I created in minetweaker. It still registers as invalid according to the log.

http://pastebin.com/wUQ7N6aB

Also, it seems that natura fences with ropes turn into eucalyptus wood.

@OreCruncher @IceDragon200 

https://github.com/OreCruncher/BetterRain/issues/96

https://github.com/OreCruncher/BetterRain/issues/79#issuecomment-186692400

IceDragon200 commented:
@Sunconure11 ItemKeySchema's track the resultant itemstacks before creating the recipe.

I'm guessing MineTweaker runs pretty late in the initialization cycle

Could you give the latest build a spin, and test the fences again?

IceDragon200 committed:
api/core: ItemKeySchemas are valid if their ore is not null

IceDragon200 committed:
core: Fixed FenceRope not accepting metadata from blocks

IceDragon200 committed:
hops: Replaced hardcoded wheat and hops items with OreItemStacks

IceDragon200 committed:
all*: Refactored all items, added description and shift to toggle booze information

IceDragon200 committed:
cellar: Added drinking events for WaterBag

IceDragon200 committed:
cellar: Added copyright header to ItemYeast and ItemWaterBag

IceDragon200 committed:
api/core,cellar: Water bags can now pick up fluid blocks

IceDragon200 committed:
core: Fixed item descriptions

IceDragon200 committed:
core/util: ItemUtils replaced homebrewed drop logic with dropPlayerItemWithRandomChoice

IceDragon200 committed:
api/cellar,apples,bees,cellar,grapes,hops,milk,rice: Refactored booze food stats

Food Stats (healAmount && saturation) have been added to BoozeEntry
Everything has been dropped to 1 heal, some saturation values have been tweaked.
Water Bag now respects Booze food stats
Water Bag now implements the AppleCore IEdible interface to show new heal and saturation values

Sunconure11 commented:
Both issues have been resolved

IceDragon200 committed:
bees,core: Added Natura Bee Boxes and ...

renamed all unlocalized names for bee boxes

IceDragon200 committed:
core: Stubbed a few missing translations

IceDragon200 committed:
bees: Removed unused import in BlockBeeBoxThaumcraft

IceDragon200 committed:
api/core: Added IMultiFluidStacks#containsFluid

IceDragon200 committed:
api/core: FluidTest and ItemTest can now handle Lists of IMultiItemStacks and ItemStacks

Even if mixed, might be a bit slower though :O

IceDragon200 committed:
core/common/inventory: InventoryProcessor now supports generic List types

This allows the handling of ItemStacks and IMultiItemStacks

IceDragon200 committed:
api/milk,milk: Cheese Vat now uses IMulti*Stacks directly instead of unrolling

IceDragon200 committed:
api: Added Inventory Tweaks api for future use

Joccob committed:
Update zh_CN.lang

IceDragon200 committed:
Merge pull request #305 from Joccob/patch-1

Update zh_CN.lang

IceDragon200 committed:
Updated CONTRIBUTORS.md

Commit:
Update gradle.properties
